src/Andre/SoulsFormats/SoulsFormats/Util/BinaryReaderEx.cs

```diff
@@ -1136,6 +1136,44 @@ public string GetShiftJIS(long offset, int length)
             return result;
         }
 
+        /// <summary>
+        /// Reads a null-terminated UTF-8 string.
+        /// </summary>
+        public string ReadUTF8()
+        {
+            List<byte> bytes = new List<byte>(64);
+            byte a = ReadByte();
+            byte b = ReadByte();
```

Comment:
Shouldn't need to be reading two bytes at once. not only is utf8 a variable length encoding from 1-4 bytes, it does guarantee 0x0 only appears for null character, ie null terminated string.
I suspect reading two bytes you'll have a bad read if you try reading a name that's 7 characters long, for example.

Reply:
Updated to read a single byte at a time and just look for the carriage return and then advance a character for the line feed, which seemed like the simplest solution.